tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -186,6 +186,11 @@ await Task.WhenAny(_flushTask, Task.Delay(TimeSpan.FromSeconds(20)))
                 await _statsAggregator.DisposeAsync().ConfigureAwait(false);
             }
 
+            if (_api is IDisposable disposableApi)
+            {
+                disposableApi.Dispose();
```

Comment:
I'm not sure `AgentWriter` should dispose an `IApi` it doesn't own (it's passed in via ctor). A caller could potentially re-use the `IApi`.